zhaquirks/tuya/ts0601_sensor.py

```diff
@@ -334,154 +327,23 @@ class TuyaTempHumiditySensorVar04(CustomDevice):
-        ),
-    }
-
-    data_point_handlers = {
-        3: "_dp_2_attr_update",
-        5: "_dp_2_attr_update",
-        15: "_dp_2_attr_update",
-    }
-
-
-class TuyaSoilSensor(CustomDevice):
-    """Tuya temp and humidity sensor (variation 03)."""
-
-    signature = {
-        # "profile_id": 260,
-        # "device_type": "0x0051",
-        # "in_clusters": ["0x0000","0x0004","0x0005","0xef00"],
-        # "out_clusters": ["0x000a","0x0019"]
-        MODELS_INFO: [
-            ("_TZE200_myd45weu", "TS0601"),
-            ("_TZE200_ga1maeof", "TS0601"),
-            ("_TZE200_9cqcpkgb", "TS0601"),
-            ("_TZE204_myd45weu", "TS0601"),
-        ],
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    TemperatureHumidityManufCluster.cluster_id,
-                ],
-                OUTPUT_CLUSTERS: [Ota.cluster_id, Time.cluster_id],
-            }
-        },
-    }
-
-    replacement = {
-        SKIP_CONFIGURATION: True,
-        ENDPOINTS: {
-            1: {
-                DEVICE_TYPE: zha.DeviceType.TEMPERATURE_SENSOR,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    SoilManufCluster,
-                    TuyaTemperatureMeasurement,
-                    TuyaSoilMoisture,
-                    TuyaPowerConfigurationCluster2AAA,
-                ],
-                OUTPUT_CLUSTERS: [Ota.cluster_id, Time.cluster_id],
-            }
-        },
-    }
-
-
-class SoilManufClusterVar02(TuyaMCUCluster):
-    """Tuya Manufacturer Cluster with Temperature and Humidity data points (variation 02)."""
-
-    dp_to_attribute: dict[int, DPToAttributeMapping] = {
-        5: DPToAttributeMapping(
-            TuyaTemperatureMeasurement.ep_attribute,
-            "measured_value",
-            converter=lambda x: x * 10,
-        ),
-        3: DPToAttributeMapping(
-            TuyaSoilMoisture.ep_attribute,
-            "measured_value",
-            converter=lambda x: x * 100,
-        ),
-        15: DPToAttributeMapping(
-            TuyaPowerConfigurationCluster2AAA.ep_attribute,
-            "battery_percentage_remaining",
-            converter=lambda x: x * 2,  # double reported percentage
-        ),
-    }
-
-    data_point_handlers = {
-        3: "_dp_2_attr_update",
-        5: "_dp_2_attr_update",
-        15: "_dp_2_attr_update",
-    }
-
-
-class TuyaSoilSensorVar02(CustomDevice):
-    """Tuya temp and humidity sensor (variation 05)."""
+(
+    TuyaQuirkBuilder("_TZE284_aao3yzhs", "TS0601")
+    .also_applies_to("_TZE284_sgabhwa6", "TS0601")
+    .temperature(5, converter=lambda x: x * 10)
```

Comment:
Also, would an extra argument like `multiplication` make sense here (with maybe `10` as default `.temperature`?
So, we could just pass `multiplication=100` if we want that. I think it's rare we need to do more complex calculations for basic entities (like temperature)(?)

Reply:
I had the same thought, the converter was the safe option, are we good with something like scale=10, then if it's something else they can add it using `.tuya_dp`?

Reply:
I think that should be fine.

Reply:
Should we just stick w/ the common multiplier and divisor like we have for other clusters?

Reply:
Hmm, that's only illuminance, right? We can just put the "exotic part" of the converter in the `. illuminance()` method.